spec/rubocop/ast/send_node_spec.rb

```diff
@@ -94,6 +94,35 @@
     end
   end
 
+  describe '#access_modifier?' do
+    context 'when inside a module' do
+      let(:send_nodes_in_module) { source.ast.children[1].children }
+
+      let(:source) { parse_source(<<-RUBY) }
+        module Foo
+          module_function
+          public
+          protected
+          private
+          custom_macro!
+          self.private
+        end
+      RUBY
+
+      it 'classifies specific macros like `private` as access modifiers' do
```

Comment:
I really prefer the (very repetitive) style used in the rest of the spec, with one `it` block per assertion, encapsulating the source. 🙂